cloud-aws-common/src/test/java/com/sequenceiq/cloudbreak/cloud/aws/common/connector/resource/AwsResourceVolumeConnectorTest.java

```diff
@@ -0,0 +1,82 @@
+package com.sequenceiq.cloudbreak.cloud.aws.common.connector.resource;
+
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.doReturn;
+
+import java.util.List;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Spy;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.mockito.stubbing.Answer;
+
+import com.sequenceiq.cloudbreak.cloud.aws.common.client.AmazonEc2Client;
+import com.sequenceiq.cloudbreak.cloud.aws.common.view.AuthenticatedContextView;
+import com.sequenceiq.cloudbreak.cloud.context.AuthenticatedContext;
+import com.sequenceiq.cloudbreak.cloud.model.CloudResource;
+import com.sequenceiq.cloudbreak.cloud.model.VolumeSetAttributes;
+
+import software.amazon.awssdk.services.ec2.model.DescribeVolumesResponse;
+import software.amazon.awssdk.services.ec2.model.VolumeState;
+
+@ExtendWith(MockitoExtension.class)
+public class AwsResourceVolumeConnectorTest {
+
+    @Mock
+    private AuthenticatedContext authenticatedContext;
+
+    @Mock
+    private CloudResource cloudResource;
+
+    @Mock
+    private VolumeSetAttributes volumeSetAttributes;
+
+    @Mock
+    private VolumeSetAttributes.Volume volume;
+
+    @InjectMocks
+    @Spy
+    private AwsResourceVolumeConnector underTest;
+
+    @Mock
+    private AuthenticatedContextView authenticatedContextView;
+
+    @Mock
+    private AmazonEc2Client amazonEc2Client;
+
+    @BeforeEach
+    public void setUp() {
+        doReturn("TEST-1").when(cloudResource).getInstanceId();
+        doReturn(volumeSetAttributes).when(cloudResource).getParameter(CloudResource.ATTRIBUTES, VolumeSetAttributes.class);
+        doReturn(List.of(volume)).when(volumeSetAttributes).getVolumes();
+        doReturn("vol-test-1").when(volume).getId();
+        doReturn(amazonEc2Client).when(underTest).getEc2Client(authenticatedContext);
+    }
+
+    @Test
+    public void testDetachVolumes() throws Exception {
+        doAnswer(new Answer<VolumeState>() {
+            private int count;
+            public VolumeState answer(InvocationOnMock invocation) {
+                if (count++ == 1) {
+                    return VolumeState.IN_USE;
+                }
+                return VolumeState.AVAILABLE;
+            }
+        }).when(underTest).getVolumeState("vol-test-1", amazonEc2Client);
+        software.amazon.awssdk.services.ec2.model.Volume volumeResponse = software.amazon.awssdk.services.ec2.model.Volume.builder()
+                .state(VolumeState.AVAILABLE).build();
+        DescribeVolumesResponse response = DescribeVolumesResponse.builder().volumes(List.of(volumeResponse)).build();
+        underTest.detachVolumes(authenticatedContext, List.of(cloudResource));
+    }
+
+    @Test
```

Comment:
Can we have additional tests where the related SDK calls throws exception?

Reply:
done